File: python/resistor-color/resistor_color.py
# The band colors are encoded as follows:
# Black: 0
# Brown: 1
# Red: 2
# Orange: 3
# Yellow: 4
# Green: 5
# Blue: 6
# Violet: 7
# Grey: 8
# White: 9

RESISTANCE_VALUES = {
    "black": 0,
    "brown": 1,
    "red": 2,
    "orange": 3,
    "yellow": 4,
    "green": 5,
    "blue": 6,
    "violet": 7,
    "grey": 8,
    "white": 9,
}


def color_code(color):
    return RESISTANCE_VALUES[color]


def colors():
    return list(RESISTANCE_VALUES.keys())

File: java/resistor-color/src/main/java/ResistorColor.java
import java.util.Arrays;
import java.util.Comparator;
import java.util.HashMap;
import java.util.Map;

class ResistorColor {
  private static final Map<String, Integer> colorValues = initializeColorValues();

  private static Map<String, Integer> initializeColorValues() {
    final Map<String, Integer> colorValues = new HashMap<String, Integer>();
    colorValues.put("black", 0);
    colorValues.put("brown", 1);
    colorValues.put("red", 2);
    colorValues.put("orange", 3);
    colorValues.put("yellow", 4);
    colorValues.put("green", 5);
    colorValues.put("blue", 6);
    colorValues.put("violet", 7);
    colorValues.put("grey", 8);
    colorValues.put("white", 9);
    return colorValues;
  }

  int colorCode(final String color) {
    return colorValues.get(color);
  }

  String[] colors() {
    final String[] colorNames = colorValues.keySet().toArray(new String[colorValues.size()]);
    Arrays.sort(
        colorNames,
        new Comparator<String>() {

          @Override
          public int compare(final String color1, final String color2) {
            return colorValues.get(color1) - colorValues.get(color2);
          }
        });

    return colorNames;
  }
}
